src/vdaf/prg.rs

```diff
@@ -232,6 +237,112 @@ impl SeedStream for SeedStreamSha3 {
     }
 }
 
+/// PrgFixedKeyAes128 as specified in [[draft-irtf-cfrg-vdaf-05]]. This PRG is NOT RECOMMENDED for
+/// general use; see Section 9 ("Security Considerations") for details.
+///
+/// This PRG combines SHA-3 and a fixed-key mode of operation for AES-128. The key is "fixed" in
+/// the sense that it is derived (using cSHAKE128) from the customization and binder strings, and
+/// depending on the application, these strings can be hard-coded. The seed is used to construct
+/// each block of input passed to a hash function built from AES-128.
+///
+/// [draft-irtf-cfrg-vdaf-05]: https://datatracker.ietf.org/doc/draft-irtf-cfrg-vdaf/05/
+#[derive(Clone, Debug)]
+#[cfg(feature = "experimental")]
+pub struct PrgFixedKeyAes128 {
+    fixed_key_deriver: CShake128,
+    base_block: Block,
+}
+
+#[cfg(feature = "experimental")]
+impl Prg<16> for PrgFixedKeyAes128 {
+    type SeedStream = SeedStreamFixedKeyAes128;
+
+    fn init(seed_bytes: &[u8; 16], custom: &[u8]) -> Self {
+        Self {
+            fixed_key_deriver: CShake128::from_core(CShake128Core::new(custom)),
+            base_block: (*seed_bytes).into(),
+        }
+    }
+
+    fn update(&mut self, data: &[u8]) {
+        Update::update(&mut self.fixed_key_deriver, data);
+    }
+
+    fn into_seed_stream(self) -> SeedStreamFixedKeyAes128 {
+        let mut fixed_key = GenericArray::from([0; 16]);
+        XofReader::read(&mut self.fixed_key_deriver.finalize_xof(), &mut fixed_key);
+        SeedStreamFixedKeyAes128 {
+            base_block: self.base_block,
+            cipher: Aes128::new(&fixed_key),
+            length_consumed: 0,
+        }
+    }
+}
+
+/// Seed stream for [`PrgFixedKeyAes128`].
+#[cfg(feature = "experimental")]
+pub struct SeedStreamFixedKeyAes128 {
+    cipher: Aes128,
+    base_block: Block,
+    length_consumed: u64,
+}
+
+#[cfg(feature = "experimental")]
+impl SeedStream for SeedStreamFixedKeyAes128 {
+    fn fill(&mut self, buf: &mut [u8]) {
+        let next_length_consumed = self.length_consumed + u64::try_from(buf.len()).unwrap();
+        let mut offset = usize::try_from(self.length_consumed % 16).unwrap();
+        let mut index = 0;
+        let mut block = Block::from([0; 16]);
+
+        // NOTE(cjpatton) We might be able to speed this up by unrolling this loop and encrypting
+        // multiple blocks at the same time via `self.cipher.encrypt_blocks()`.
+        for block_counter in self.length_consumed / 16..(next_length_consumed / 15) + 16 {
```

Comment:
This has a typo. I presume the test vector test didn't catch it because we're not sampling much data out of the PRG, so all calls fit within 16 blocks. (as before, extra outer loop iterations wind up as no-ops because `read` is 0 once the buffer's full)
```suggestion
        for block_counter in self.length_consumed / 16..(next_length_consumed + 15) / 16 {
```